examples/pi-home-router/src/arp/arp_generator.rs

```diff
@@ -0,0 +1,114 @@
+use crate::arp::ArpOp::Request;
+use crate::arp::{ArpFrame, ArpTable, IPV4_PROTOCOL_TYPE};
+use crate::types::InterfaceAnnotated;
+use route_rs_packets::{EthernetFrame, Ipv4Packet, Ipv6Packet, MacAddr};
+use route_rs_runtime::processor::Processor;
+use std::convert::TryFrom;
+use std::net::IpAddr;
+
+pub(crate) struct ArpGenerator {
+    // Mapping between (protocol type, protocol address) -> 48-bit Ethernet address
+    arp_table: ArpTable,
+}
+
+impl ArpGenerator {
+    pub fn new() -> Self {
+        ArpGenerator {
+            arp_table: ArpTable::new(),
+        }
+    }
+}
+
+impl Processor for ArpGenerator {
+    type Input = InterfaceAnnotated<EthernetFrame>;
+    type Output = InterfaceAnnotated<EthernetFrame>;
+
+    ///
+    /// As a packet is sent down through the network layers, routing determines the protocol address
```

Comment:
This seems like it's coming from an RFC or something, can you cite the source?

Reply:
I stuck it in the PR description, but I can put it with the code as well. `ArpHandler` and `ArpGenerator` both have paraphrased descriptions from the RFC.